src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -172,4 +176,16 @@ public String getOptionalArrayTypeName() {
   public String getDynamicReturnTypeName(Method method, MethodConfig methodConfig) {
     return getNotImplementedString("SurfaceNamer.getDynamicReturnTypeName");
   }
+
+  public String getApiSettingsClassName(Interface service) {
+    return className(Name.upperCamel(service.getSimpleName(), "Settings"));
+  }
+
+  public String getTestClassName(Interface service) {
+    return className(Name.upperCamel(service.getSimpleName(), "Test"));
+  }
+
+  public String getMockServiceClassName(Interface service) {
+    return className(Name.upperCamel("Mock", service.getSimpleName()));
+  }
```

Comment:
I'm a little concerned about an explosion of methods in this class that each are only used in one or two places in the code. Seems like this special case logic would be better defined in the file where it's used, unless it truly has broad applicability.